rmc/utils/generic.py

```diff
@@ -289,50 +289,68 @@ def get_ref_aa(
     :param extra_aa_map: Dictionary mapping any amino acid one letter codes to three letter codes.
         Designed to capture any amino acids not present in `ACID_NAMES_PATH`.
         Default is {"*": "Ter", "U": "Sec"}.
-    :param aa_to_remove: Any amino acid one letter codes to remove. Default is "X".
-    :return: HT with one reference amino acid annotated per locus/transcript if `collect_all_aa` is False.
-        HT grouped by transcript with a list of all reference amino acids in that transcript if `collect_all_aa` is True.
+    :param aa_to_remove: Any amino acid one letter codes to remove. Default is {"X"}.
+    :return: HT with one reference amino acid annotated per locus/transcript.
     """
     logger.info("Mapping amino acid one letter codes to three letter codes...")
     aa_map = get_aa_map()
     # Add any extra mappings into amino acid mapping dict
     if extra_aa_map:
-        aa_map = aa_map.update(extra_aa_map)
-    ht = ht.annotate(ref_aa_1_letter=ht.amino_acids.split("/")[0])
-    ht = ht.annotate(
-        ref_aa=hl.literal(aa_map).get(ht.ref_aa_1_letter, ht.ref_aa_1_letter)
-    )
+        aa_map.update(extra_aa_map)
+    aa_map = hl.literal(aa_map)
+    ht = ht.annotate(ref_aa=aa_map.get(ht.ref_aa_1_letter, ht.ref_aa_1_letter))
     if aa_to_remove:
-        ht = ht.filter(ht.ref_aa != aa_to_remove)
+        ht = ht.filter(hl.literal(aa_to_remove).contains(ht.ref_aa))
+
+    # Select fields and checkpoint
+    ht = ht.select("ref_aa", "aa_start_num", "transcript")
+    ht = ht.checkpoint(
+        f"{TEMP_PATH_WITH_FAST_DEL}/rmc/ref_amino_acids.ht",
+        _read_if_exists=not overwrite_temp,
+        overwrite=overwrite_temp,
+    )
+
+    # Check if any there are any ref amino acids in HT that aren't in aa_map
+    ref_aa_check = ht.aggregate(hl.agg.collect_as_set(ht.ref_aa))
+    ref_aa_check = ref_aa_check.difference(set(hl.eval(aa_map).keys()))
+    if len(ref_aa_check) != 0:
+        logger.warning(
+            "The following reference amino acids were not mapped to three letter"
+            " codes: %s",
+            ref_aa_check,
+        )
 
     # Double check that protein start always equals protein end
     protein_num_check = ht.aggregate(
```

Comment:
Should we maybe make this check and the one on reference AAs into an expression that we write out + read if exists? Like in the LOFTEE aggs of misbad calculation. So we don't have to recompute every time we run.

Reply:
good idea, done